lib/ansible/module_utils/urls.py

```diff
@@ -1277,8 +1278,12 @@ def open(self, method, url, data=None, headers=None, use_proxy=None,
             request.add_header('If-Modified-Since', tstamp)
 
         # user defined headers now, which may override things we've set above
+        unredirected_headers = unredirected_headers or []
         for header in headers:
-            request.add_header(header, headers[header])
+            if header in unredirected_headers:
+                request.add_unredirected_header(header, headers[header])
```

Comment:
This is interesting, I didn't even know this was a thing. I have no problems with it.

Reply:
Yea there were a few Python ML pages where people complained it wasn't the default for headers like `Authorization` and `Cookie` and this was their recommended way of doing it.